File: playground/vote.rb
class Vote
  attr_accessor :question
  attr_reader :results

  def initialize(q)
    @question = q
    @results  = {}
  end

  def cast(answer)
    @results[answer] ||= 0
    @results[answer] += 1
  end

  def total
    sum = 0
    @results.values.each {|n| sum += n }
    sum
  end
end

File: playground/vote.js
var vote = {
  results: {},
  question: "Who should win?"
};

vote.cast = function (n) {
  if (!this.results[n]) {
    this.results[n] = 0;
  }
  this.results[n]++;
  return this.results;
};

vote.cast("Bernie");
vote.cast("Bernie");
vote.cast("Bernie");
vote.cast("Bernie");
vote.cast("Trump");
vote.cast("Hillary");
vote.cast("Hillary");
vote.cast("Hillary");

console.log(vote.results);
